rcsbsearchapi/search.py

```diff
@@ -1502,11 +1736,50 @@ def _make_params(self, start=0):
-                query_dict["request_options"]["facets"] = [self.facets.to_dict()]
+
+        if self.request_options:
+            for request_option in self.request_options:
+                if isinstance(request_option, Sort):
+                    if "sort" not in query_dict["request_options"]:
+                        query_dict["request_options"]["sort"] = []
+                    query_dict["request_options"]["sort"].append(request_option.to_dict())
+
+                if isinstance(request_option, GroupBy):
+                    if "group_by" in query_dict["request_options"]:
+                        raise ValueError("Multiple GroupBy request options are not allowed")
+                    query_dict["request_options"]["group_by"] = request_option.to_dict()
+
+                if isinstance(request_option, GroupByReturnType):
+                    if "group_by_return_type" in query_dict["request_options"]:
+                        raise ValueError("Multiple GroupByReturnType request options are not allowed")
+                    query_dict["request_options"]["group_by_return_type"] = request_option.group_by_return_type
+
+                if isinstance(request_option, ScoringStrategy):
+                    if "scoring_strategy" in query_dict["request_options"]:
+                        raise ValueError("Multiple ScoringStrategy request options are not allowed")
+                    query_dict["request_options"]["scoring_strategy"] = request_option.scoring_strategy
+
+                if isinstance(request_option, ReturnCounts):
+                    if "return_counts" in query_dict["request_options"]:
+                        raise ValueError("Multiple ReturnCounts request options are not allowed")
+                    query_dict["request_options"]["return_counts"] = request_option.return_counts
+
+                    # return_counts can't be used with paginate
+                    if query_dict["request_options"]["paginate"]:
+                        query_dict["request_options"].pop("paginate")
+
+                if isinstance(request_option, ReturnExplainMetadata):
+                    if "return_counts" in query_dict["request_options"]:
+                        raise ValueError("Multiple ReturnCounts request options are not allowed")
+                    query_dict["request_options"]["return_explain_metadata"] = request_option.return_explain_metadata
+
+                if isinstance(request_option, Facet) or isinstance(request_option, FilterFacet) or isinstance(request_option, TerminalFilter):
+                    if "facets" not in query_dict["request_options"]:
+                        query_dict["request_options"]["facets"] = []
+                    if isinstance(request_option, list):
+                        query_dict["request_options"]["facets"] += [facet.to_dict() for facet in self.request_option]
+                    else:
+                        query_dict["request_options"]["facets"].append(request_option.to_dict())
```

Comment:
@ivana-truong I see that you removed this section. Is it no longer needed? All the better if it isn't needed, but I thought I remember you saying that at least part of it would still be required...